From 675cdda0b5ed02a5020299ea9b900e2cc6231790 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mathieu=20Dupr=C3=A9?= <mathieu.dupre@savoirfairelinux.com>
Date: Fri, 27 Nov 2020 10:12:04 +0100
Subject: [PATCH] Add LDFLAGS when linking cython libraries

LDFLAGS variable must be passed to the linker during the link step.
It was no the cause for the libraries made with cython.
---
 cmake/modules/Distutils.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/modules/Distutils.cmake b/cmake/modules/Distutils.cmake
index b2f4223e..3f9f2222 100644
--- a/cmake/modules/Distutils.cmake
+++ b/cmake/modules/Distutils.cmake
@@ -54,7 +54,7 @@ function(distutils_add_cython_module name src)
     CXX="${PY_CXX}"
     LDSHARED="${PY_LDSHARED}"
     OPT=\"-DNDEBUG -g -fwrapv -O2 -w\"
-    LDFLAGS=-L${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
+    LDFLAGS=-L${CMAKE_LIBRARY_OUTPUT_DIRECTORY} ${CMAKE_SHARED_LINKER_FLAGS_INIT}
     CYTHON_BUILD_DIR=${CMAKE_CURRENT_BINARY_DIR}
     CEPH_LIBDIR=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
     ${PYTHON${PYTHON_VERSION}_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/setup.py
@@ -75,7 +75,7 @@ function(distutils_install_cython_module name)
     set(ENV{CPPFLAGS} \"-iquote${CMAKE_SOURCE_DIR}/src/include
                         -D'void0=dead_function\(void\)' \
                         -D'__Pyx_check_single_interpreter\(ARG\)=ARG \#\# 0'\")
-    set(ENV{LDFLAGS} \"-L${CMAKE_LIBRARY_OUTPUT_DIRECTORY}\")
+    set(ENV{LDFLAGS} \"-L${CMAKE_LIBRARY_OUTPUT_DIRECTORY} ${CMAKE_SHARED_LINKER_FLAGS_INIT}\")
     set(ENV{CYTHON_BUILD_DIR} \"${CMAKE_CURRENT_BINARY_DIR}\")
     set(ENV{CEPH_LIBDIR} \"${CMAKE_LIBRARY_OUTPUT_DIRECTORY}\")
 
-- 
2.17.1

